extensions/tls-registry/deployment/src/main/java/io/quarkus/tls/CertificatesProcessor.java

```diff
@@ -15,13 +17,26 @@
 import io.quarkus.deployment.builditem.ShutdownContextBuildItem;
 import io.quarkus.runtime.configuration.ConfigurationException;
 import io.quarkus.tls.runtime.CertificateRecorder;
+import io.quarkus.tls.runtime.KeyStoreProvider;
 import io.quarkus.tls.runtime.LetsEncryptRecorder;
+import io.quarkus.tls.runtime.TrustStoreProvider;
 import io.quarkus.tls.runtime.config.TlsConfig;
 import io.quarkus.vertx.deployment.VertxBuildItem;
 import io.quarkus.vertx.http.deployment.spi.RouteBuildItem;
+import io.smallrye.common.annotation.Identifier;
 
 public class CertificatesProcessor {
 
+    @BuildStep
+    public AdditionalBeanBuildItem additionalBeans() {
+        return AdditionalBeanBuildItem.builder().addBeanClass(Identifier.class).build();
```

Comment:
Identifier is a qualifier. You should use the type name here.

Reply:
@cescoffier do you mean I should use the literal class name instead of importing and using the class directly? The guidance [here](https://quarkus.io/guides/cdi-integration#use-case-my-annotation-is-not-recognized-as-a-qualifier-or-an-interceptor-binding) is not clear about how qualifiers should be registered.

Reply:
I don't think you need the type here, they are users who provide these beans, not the framework.  The `io.smallrye.common.annotation.Identifier` is annotated with `jakarta.inject.Qualifier` so I don't think you need this build step at all.

Reply:
>  The io.smallrye.common.annotation.Identifier is annotated with jakarta.inject.Qualifier so I don't think you need this build step at all.

You do need to make sure the annotation class is part of the application index, i.e. is contained in a bean archive (or in an archive that is indexed explicitely using `quarkus.index-dependency` entries) or registered as an additional bean class. In this particular case, it should not be necessary because `smallrye-common-annotation` contains a Jandex index and we also include app archives that are not bean archives but contain scopes, qualifiers or interceptor bindings.

Reply:
Good point about being part of the index, I didn't realize that because I inspected where else is it used in Quarkus and noone seemed to register anything, just use it.

Reply:
> registered as an additional bean class

bit surprising, I would expect it should be registered with `QualifierRegistrarBuildItem`. Is a qualifier bean or why it works this way @mkouba ?

Reply:
> ...I don't think you need this build step at all.

@michalvavrik you are right, thanks!

I had this build step from when I had a custom qualifier and just refactored it to use `@Identifier` when I saw @cescoffier 's suggestion. I will remove it.